File: Platform.FoodMe.Admin/Platform.FoodMe.Admin/src/app/routes/ReferentialData/Models/User Data/RoleVM.ts
import { Status } from '@shared/Models/LocationData/CountryDto';
import { RoleClaimVM } from './RoleClaimVM';

export class RoleVM{
  id: string ;
  name: string ;
  status : Status = Status.isActive;
  creatorUserId : string ;
  creationTime: Date;
  lastModifierUserId : string;
  lastModificationTime: Date;
  claims : Array<RoleClaimVM> = [];
  DeleterUserId :string;
  DeletionTime :Date;
}

export class CreateRoleVM{
  name: string ;
  claims : Array<RoleClaimVM> = [];
}

File: Platform.FoodMe.Admin/Platform.FoodMe.Admin.ViewModels/ViewModels/AuthenticationData/RoleVM.cs
using Platform.Shared.Enum;

namespace Platform.FoodMe.Admin.ViewModels.ViewModels.AuthenticationData;

public class RoleVM
{
    public string Id { get; set; }

    public string Name { get; set; }

    public Status Status { get; set; }

    public DateTime CreationTime { get; set; }

    public string CreatorUserId { get; set; }

    public string DeleterUserId { get; set; }

    public DateTime? DeletionTime { get; set; }

    public string? LastModifierUserId { get; set; }

    public DateTime? LastModificationTime { get; set; }

    public IList<RoleClaimVM> Claims { get; set; }
}

public class CreateRoleVM
{
    public string Name { get; set; }
    public IList<RoleClaimVM> Claims { get; set; }
}
